force-app/main/default/classes/AN_AutoNumberService.cls

```diff
@@ -202,6 +202,9 @@ public class AN_AutoNumberService {
         }
 
         public String getFormattedNumber(Integer num) {
+            // A change in API version 51 and later causes this to fail, so we're keeping 
+            // this Class at 50 for now.
+            // {000} is no longer a valid pattern for the first match when using String.format
```

Comment:
Hmm. Technically `{000}` should have never been supported. In a format string the number zero represents the first parameter in the list sent to `String.format()`. It's very different than the format of an AutoNumber field. It looks like the the number of zeros is used to control the result of `getNumberLength()`, effectively trying to simulate how an auto-number field works.

So, maybe the fix here (to allow us to move this to api 53) is to first call `getNumberLength()` and then replace the format pattern in the string to just `{0}`. That would allow for both the simulation and format logic to work:
* `'ABCZZZ-{000}'` ==> `'ABCZZZ-{0}'`

Reply:
Works for me, I'll get this change (as well as the new Classes) in today.

Reply:
Alright, these changes are in, tests pass, but I haven't done any actual testing with the AN_AutoNumberService yet